contracts/test/integration/vip/vip8.sol

```diff
@@ -0,0 +1,133 @@
+        public
+        pure
+        override
+        returns (TimelockSimulation.action[] memory proposal)
+    {
+        proposal = new TimelockSimulation.action[](4);
+
+        proposal[0].target = MainnetAddresses.VOLT_FEI_PSM;
+        proposal[0].value = 0;
+        proposal[0].arguments = abi.encodeWithSignature("pauseRedeem()");
+        proposal[0].description = "Pause redemptions on the FEI PSM";
+
+        proposal[1].target = MainnetAddresses.FEI;
+        proposal[1].value = 0;
+        proposal[1].arguments = abi.encodeWithSignature(
+            "approve(address,uint256)",
+            MainnetAddresses.MAKER_ROUTER,
+            2_400_000e18
+        );
+        proposal[1].description = "Timelock approves router to spend FEI";
+
+        proposal[2].target = MainnetAddresses.MAKER_ROUTER;
+        proposal[2].value = 0;
+        proposal[2].arguments = abi.encodeWithSignature(
+            "swapAllFeiForDai(address)",
+            MainnetAddresses.VOLT_DAI_PSM
+        );
+        proposal[2].description = "Swaps FEI for DAI proceeds sent to DAI PSM";
+
+        proposal[3].target = MainnetAddresses.FEI;
+        proposal[3].value = 0;
+        proposal[3].arguments = abi.encodeWithSignature(
+            "approve(address,uint256)",
+            MainnetAddresses.MAKER_ROUTER,
+            0
+        );
+        proposal[3]
+            .description = "Timelock revokes router approval to spend FEI";
+    }
+
+    function mainnetSetup() public override {
+        vm.startPrank(MainnetAddresses.GOVERNOR);
+
+        PCVGuardian(MainnetAddresses.PCV_GUARDIAN).addWhitelistAddress(
+            MainnetAddresses.MAKER_ROUTER
+        );
+
+        PCVGuardian(MainnetAddresses.PCV_GUARDIAN)
+            .withdrawAllERC20ToSafeAddress(
+                MainnetAddresses.VOLT_FEI_PSM,
+                MainnetAddresses.FEI
+            );
+
+        uint256 feiBalance = IERC20(MainnetAddresses.FEI).balanceOf(
+            MainnetAddresses.GOVERNOR
+        );
+
+        IERC20(MainnetAddresses.FEI).safeTransfer(
+            MainnetAddresses.TIMELOCK_CONTROLLER,
+            feiBalance
+        );
+
+        uint256 voltBalance = IERC20(MainnetAddresses.VOLT).balanceOf(
+            MainnetAddresses.GOVERNOR
+        );
+
+        IERC20(MainnetAddresses.VOLT).safeTransfer(
+            MainnetAddresses.TIMELOCK_CONTROLLER,
+            voltBalance
+        );
+        vm.stopPrank();
+    }
+
+    function mainnetValidate() public override {
+        assertTrue(
+            PegStabilityModule(MainnetAddresses.VOLT_FEI_PSM).redeemPaused()
+        );
+
+        assertEq(
+            IERC20(MainnetAddresses.FEI).balanceOf(
+                MainnetAddresses.VOLT_FEI_PSM
+            ),
+            0
+        );
+
+        assertEq(
+            IERC20(MainnetAddresses.FEI).allowance(
+                MainnetAddresses.TIMELOCK_CONTROLLER,
+                MainnetAddresses.MAKER_ROUTER
+            ),
+            0
+        );
```

Comment:
we should assert that the DAI PSM has the proper amount of DAI in it after the proposal.

Reply:
We won't know what this exact amount is at the time of execution as we don't know how much FEI is being converted, although since minting has been paused we only know that the PSM won't receive any more FEI so it'll be a max of ~2.3m, it could be less than that, so we can't accurately assert the DAI PSM balance

Reply:
we can assert it is greater than 0 and more than 2m which we should do